views/web_views.py

```diff
@@ -270,6 +276,75 @@ def serve_origin_js(version):
     req = requests.get(url, stream=True)
     return Response(stream_with_context(req.iter_content(chunk_size=2048)), content_type="text/javascript")
 
+YOUTUBE_CONFIG = {
+  "web":
+  {
+    "client_id": constants.YOUTUBE_CLIENT_ID,
+    "project_id": constants.YOUTUBE_PROJECT_ID,
+    "auth_uri": "https://accounts.google.com/o/oauth2/auth",
+    "token_uri": "https://www.googleapis.com/oauth2/v3/token",
+    "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
+    "client_secret": constants.YOUTUBE_CLIENT_SECRET,
+    "redirect_uris":[constants.YOUTUBE_REDIRECT_URL]
+  }
+}
+
+SCOPES = ['https://www.googleapis.com/auth/youtube.readonly']
+
+@app.route('/youtube')
+def youtube():
+  if not (constants.YOUTUBE_TOKEN and constants.YOUTUBE_REFRESH_TOKEN):
+    return redirect('/youtube/authorize')
+
+  return redirect(url_for('index', lang_code=get_locale()))
+
+@app.route('/youtube/authorize')
+def authorize():
+  flow = google_auth_oauthlib.flow.Flow.from_client_config(
+      client_config=YOUTUBE_CONFIG, scopes=SCOPES)
+  flow.redirect_uri = url_for('oauth2callback', _external=True)
+  authorization_url, state = flow.authorization_url(
+      access_type='offline',
+      include_granted_scopes='true')
+
+  session['state'] = state
+
+  return redirect(authorization_url)
+
+
+@app.route('/youtube/oauth2callback')
+def oauth2callback():
+  state = session['state']
+  flow = google_auth_oauthlib.flow.Flow.from_client_config(
+      client_config=YOUTUBE_CONFIG, scopes=SCOPES, state=state)
+  flow.redirect_uri = url_for('oauth2callback', _external=True)
+
+  authorization_response = request.url
+  flow.fetch_token(authorization_response=authorization_response)
+
+  #save the token and refresh_token in the credentials as environment variables
+  credentials = flow.credentials
+
+  print("YOUTUBE TOKEN", credentials.token)
+  print("YOUTUBE REFRESH_TOKEN", credentials.refresh_token)
```

Comment:
@joshfraser is printing to the console like this okay (to then throw into env vars)?